### DISCUSSION TITLE
React App Firestore create error "POST Error: 400, Not Found"

### DISCUSSION THREAD

**Discussion by sharonanderson-dev** [community/community#36504] [Other Feature Feedback, Questions, & Ideas] [answered]
I was running an react app linked to firebase and uses `auth` and `firestore`. I don't know if this is related to the recent bump of firebase-tools to 6.0 or not. But I'm unable to send my files to `firestore`. 

My React Version
```
8.14.0
```

My Firebase Version
```
7.20.0
```
Here is the Screenshot of that Error
![image](https://user-images.githubusercontent.com/92415144/196320537-b9b68a8b-bf1d-42cd-9d69-8bc77ddfda8b.png)

How to fix this?

**Comment by santosharron** [comment, marked as answer]
The problem is with the firebase data which you're importing. How did you link your firebase to your app?

**Reply by sharonanderson-dev** [reply]
I linked my `firebase.js` file into my main `app()` and my `firebase.js` would look something like this

```javascript
import firebase from "firebase/app"
import "firebase/auth"
import "firebase/firestore";

const app = firebase.initializeApp({
  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.REACT_APP_FIREBASE_DATABASE_URL,
  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.REACT_APP_FIREBASE_APP_ID
})

export const auth = app.auth()
export default app
```

**Reply by santosharron** [reply]
Alright sounds good, How did you initialize it?

**Reply by sharonanderson-dev** [reply]
I had imported `firebase` and `uuid` from the node_modules
```javascript
import firebase from "../firebase";
import { v4 as uuidv4 } from "uuid";
```

and used `addUser()` function to push my data
```javascript
const ref = firebase.firestore().collection("users")

  function addUser(newUser) {
    ref
      .doc(newUser.id)
      .set(newUser)
      .then(() => {
        setUser((prev) => [newUser, ...prev]);
      })
      .catch((err) => {
        console.error(err);
      });
  }
```
and used `handleSubmit(e)` event to callback this function 
```javascript
addUser({ name, email, id: uuidv4() })
```

**Reply by santosharron** [reply]
Alright, So Far So Good. I would recommend you to delete the `node_modules` folder and install the dependencies again using
```
npm install
```
or 
```
yarn
```
depending on your project and try to `POST` the API again.

**Reply by sharonanderson-dev** [reply]
Thanks, it worked!
